test/e2e/ha/upgrade_core_contrail_services_test.go

```diff
@@ -0,0 +1,470 @@
+				rmqContainer.Image = targetImage
+				return nil
+			})
+			require.NoError(t, err)
+			t.Run("then Rabbitmq resource has updated image", func(t *testing.T) {
+				err := wait.Contrail{
+					Namespace:     namespace,
+					Timeout:       5 * time.Minute,
+					RetryInterval: retryInterval,
+					Client:        f.Client,
+					Logger:        log,
+				}.ForPodImageChange(f.KubeClient, "rabbitmq", targetImage, "rabbitmq")
+				require.NoError(t, err)
+			})
+
+			t.Run("then Rabbitmq StatefulSet should be updated and ready", func(t *testing.T) {
+				assert.NoError(t, w.ForReadyStatefulSet("upgradetest-rabbitmq-rabbitmq-statefulset", replicas))
+			})
+		})
+
+		t.Run("when Cassandra is updated with newer image", func(t *testing.T) {
+			t.Skip()
+			_, err := controllerutil.CreateOrUpdate(context.Background(), f.Client.Client, cluster, func() error {
+				csContainer := utils.GetContainerFromList("cassandra", cluster.Spec.Services.Cassandras[0].Spec.ServiceConfiguration.Containers)
+				csContainer.Image = "registry:5000/common-docker-third-party/contrail/cassandra:" + intendedVersionMap["cassandra"]
+				return nil
+			})
+			require.NoError(t, err)
+			t.Run("then Cassandra StatefulSet should be updated and ready", func(t *testing.T) {
+				assert.NoError(t, w.ForReadyStatefulSet("upgradetest-cassandra-cassandra-statefulset", replicas))
+			})
+		})
+
+		t.Run("when Config is updated with newer image", func(t *testing.T) {
+			instance := &contrail.Manager{}
+			err := f.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster1", Namespace: namespace}, instance)
+			assert.NoError(t, err)
+			targetImage := "registry:5000/contrail-nightly/contrail-controller-config-api:" + intendedVersionMap["cemVersion"]
+			apiContainer := utils.GetContainerFromList("api", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			devicemanagerContainer := utils.GetContainerFromList("devicemanager", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			dnsmasqContainer := utils.GetContainerFromList("dnsmasq", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			schematransformerContainer := utils.GetContainerFromList("schematransformer", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			servicemonitorContainer := utils.GetContainerFromList("servicemonitor", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			analyticsapiContainer := utils.GetContainerFromList("analyticsapi", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			collectorContainer := utils.GetContainerFromList("collector", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			queryengineContainer := utils.GetContainerFromList("queryengine", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
+			//statusmonitorContainer := utils.GetContainerFromList("statusmonitor", instance.Spec.Services.Config.Spec.ServiceConfiguration.Containers)
```

Comment:
Similar as in other places - if this it commented out due to a bug, I would put a comment

Reply:
Done